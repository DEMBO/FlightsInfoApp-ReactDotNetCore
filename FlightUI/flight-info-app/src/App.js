import './App.css'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import React, { useState, useEffect } from 'react'
import 'react-tabs/style/react-tabs.css'
import Flights from './components/Flights/Flights'
import Passengers from './components/Passengers/Passengers'
import API from './api'

function App () {
  const [flights, setFlights] = useState([])

  const getFlights = async () => {
    const flights = await API.getFlights()
    setFlights(flights)
  }

  useEffect(() => {
    async function init () {
      await getFlights()
    }

    init()
  }, [])

  const flightNumbers = flights.map((f) => f.flightNumber)

  return (
    <div className="App">
      <h1>Flight Info App</h1>
      <Tabs>
        <TabList>
          <Tab>Flights</Tab>
          <Tab>Passengers</Tab>
        </TabList>
        <TabPanel>
          <Flights
            flights={flights}
            getFlights={getFlights}
            setFlights={setFlights}
          />
        </TabPanel>
        <TabPanel>
          {flightNumbers && flightNumbers.length > 0
            ? (
            <Passengers flightNumbers={flightNumbers} />
              )
            : (
            <h2>Add at least one flight first.</h2>
              )}
        </TabPanel>
      </Tabs>
    </div>
  )
}

export default App
