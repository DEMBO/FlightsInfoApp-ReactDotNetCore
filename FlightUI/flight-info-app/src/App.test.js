import App from './App'
import { render, cleanup, waitFor } from '@testing-library/react'
import ShallowRenderer from 'react-shallow-renderer'
import API from './api'

afterEach(() => cleanup())

test('<App /> Renders correctly', () => expect((new ShallowRenderer()).render(<App />)).toMatchSnapshot())

test('<App /> Gets flights', async () => {
  API.getFlights = jest.fn()
  API.getFlights.mockReturnValue([
    { flightNumber: 1, destination: 'one' },
    { flightNumber: 2, destination: 'two' }
  ])

  render(<App />)

  await waitFor(() => { })

  expect(API.getFlights).toHaveBeenCalled()
  expect(document.getElementsByClassName('flight')).toHaveLength(2)
}
)
