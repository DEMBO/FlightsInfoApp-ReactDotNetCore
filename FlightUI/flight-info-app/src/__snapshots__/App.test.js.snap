// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Flight Info App
  </h1>
  <Tabs>
    <TabList>
      <Tab>
        Flights
      </Tab>
      <Tab>
        Passengers
      </Tab>
    </TabList>
    <TabPanel>
      <Flights
        flights={Array []}
        getFlights={[Function]}
        setFlights={[Function]}
      />
    </TabPanel>
    <TabPanel>
      <h2>
        Add at least one flight first.
      </h2>
    </TabPanel>
  </Tabs>
</div>
`;
